third_task/src/book.ts

```diff
@@ -0,0 +1,36 @@
+import { Item } from './item';
+import { Iterator } from './pagesIterable';
+import { Pages } from './pages';
+
+export class Book extends Iterator(Item) {
+  pages: Pages;
+  title: string;
+  author: string;
+
+  constructor(title: string, author: string, pages: Pages) {
+    super();
+    this.pages = pages;
+    this.title = title;
+    this.author = author;
+  }
+
+  setTitle(title) {
+    this.title = title;
+  }
+
+  setAuthor(author) {
+    this.author = author;
+  }
+
+  getTitle(): string {
+    return this.title;
+  }
+
+  getAuthor(): string {
+    return this.author;
+  }
+
+  toString(): string {
+    return `Book: ${this.title} by ${this.author} with number of pages: ${this.pages.length}, `;
```

Comment:
The current implementation will not pass unit tests.
From unit test
```typescript
      expect(page.toString()).toEqual(`Book: Harry Potter by J. K. Rowling with number of pages: 4, here is page with text #${counter} and it\'s material is simple paper`);
```

Reply:
the second part after pages is taken from the pages class. It works